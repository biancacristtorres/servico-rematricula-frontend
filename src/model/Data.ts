export class Data {
  public data!: Date;
  public horaInicio!: Date;
  public horaFim!: Date;
}
