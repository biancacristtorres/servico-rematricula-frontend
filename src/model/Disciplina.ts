import { SituacaoDaDisciplina } from '@/common/enum/SituacaoDaDisciplina';
import { Turma } from './Turma';

export class Disciplina {
  public codigo: number;
  public nome: string;
  public situacao: SituacaoDaDisciplina;
  public modulo: string;
  public eDisciplinaEletiva: boolean;
  public possuiEquivalenciaDoisPorUm!: boolean;
  public turmas: Turma[] = [];
  public turmasEquivalentes: Turma[] = [];
  public turmasEspeciais: Turma[] = [];

  constructor(codigo: number, nome: string, situacao: SituacaoDaDisciplina,
              modulo: string, eDisciplinaEletiva: boolean) {
    this.codigo = codigo;
    this.nome = nome;
    this.situacao = situacao;
    this.modulo = modulo;
    this.eDisciplinaEletiva = eDisciplinaEletiva;
  }
}
