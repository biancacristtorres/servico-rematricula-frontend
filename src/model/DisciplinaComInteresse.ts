export class DisciplinaComInteresse {
  public codigo?: number;
  public nome?: string;
  public interesseManifestado?: boolean;

  constructor(codigo: number, nome: string, interesseManifestado: boolean) {
    this.codigo = codigo;
    this.nome = nome;
    this.interesseManifestado = interesseManifestado;
  }
}
