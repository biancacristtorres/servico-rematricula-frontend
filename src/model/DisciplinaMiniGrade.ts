export class DisciplinaMiniGrade {
  public codigoDisciplina?: number;
  public nomeDisciplina?: string;
}
