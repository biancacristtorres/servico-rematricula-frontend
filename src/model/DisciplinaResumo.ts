import { StatusDaDisciplina } from '@/common/enum/StatusDaDisciplina';

export class DisciplinaResumo {
  public codigoDisciplina?: number;
  public nomeDisciplina?: string;
  public descricaoTurma?: string;
  public status?: StatusDaDisciplina;
  public ePorData?: boolean;
  public campus?: string;
  public descricaoTipoCargaHoraria?: string;
  public datas?: string[];
  public nomeDisciplinaEquivalente?: string;
}
