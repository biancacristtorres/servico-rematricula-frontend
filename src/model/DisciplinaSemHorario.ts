export class DisciplinaSemHorario {
  public nomeDisciplina?: string;
  public nomeDisciplinaEquivalente?: string;
}
