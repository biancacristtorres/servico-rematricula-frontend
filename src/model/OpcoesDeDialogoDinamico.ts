import { ModelDeDialogoDinamico } from './ModelDeDialogoDinamico';

export class OpcoesDeDialogoDinamico {
  public componente!: string;
  public titulo!: string;
  public textoConfirmar!: string;
  public textoCancelar!: string;
  public modelDoComponente!: ModelDeDialogoDinamico;
  public exibir: boolean = true;
}
