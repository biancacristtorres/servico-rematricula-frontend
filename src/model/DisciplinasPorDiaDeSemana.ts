import { DisciplinaResumo } from './DisciplinaResumo';

export class DisciplinasPorDiaDeSemana {
  [index: number]: DisciplinaResumo[];
  public 2: DisciplinaResumo[];
  public 3: DisciplinaResumo[];
  public 4: DisciplinaResumo[];
  public 5: DisciplinaResumo[];
  public 6: DisciplinaResumo[];
  public 7: DisciplinaResumo[];
}
