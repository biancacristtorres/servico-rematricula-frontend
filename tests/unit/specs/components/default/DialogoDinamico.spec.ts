import {
  DialogoDinamicoFactory,
} from '../../../factory/components/DialogoDinamicoFactory';
import { OpcoesDeDialogoDinamico } from '@/model/OpcoesDeDialogoDinamico';

describe('Componente de Diálogo Dinâmico', () => {
  const factory = new DialogoDinamicoFactory();
  let opcoes: OpcoesDeDialogoDinamico;

  beforeEach(() => {
    opcoes = new OpcoesDeDialogoDinamico();
    opcoes.componente = '../../tests/unit/doubles/dummies/ComponenteDinamicoDummy.vue';
  });

  it('Renderiza um diálogo com um componente dinâmico', async (done) => {
    factory.dadoAProp(opcoes);
    await factory.montarComponente();

    const componenteDummy = factory.componente.find('#componente-dinamico-dummy');
    expect(componenteDummy.exists()).toBeTruthy();
    done();
  });

  it('Emite um evento para confirmar', async (done) => {
    opcoes.textoConfirmar = 'OK';
    factory.dadoAProp(opcoes);

    await factory.montarComponente();
    factory.componente.find('#confirmar').trigger('click');

    expect(factory.oEventoConfirmarFoiEmitido).toBeTruthy();
    done();
  });

  it('Emite um evento para rejeitar', async (done) => {
    opcoes.textoCancelar = 'CANCELAR';
    factory.dadoAProp(opcoes);

    await factory.montarComponente();
    factory.componente.find('#cancelar').trigger('click');

    expect(factory.oEventoCancelarFoiEmitido).toBeTruthy();
    done();
  });

  it('Renderiza um botao de confirmação', async (done) => {
    opcoes.textoConfirmar = 'OK';
    factory.dadoAProp(opcoes);
    await factory.montarComponente();

    const botaoDeConfirmacao = factory.componente.find('#confirmar');
    expect(botaoDeConfirmacao.exists()).toBeTruthy();
    expect(botaoDeConfirmacao.text()).toEqual('OK');
    done();
  });

  it('Não renderiza um botao de confirmação', async (done) => {
    factory.dadoAProp(opcoes);
    await factory.montarComponente();

    const botaoDeConfirmacao = factory.componente.find('#confirmar');
    expect(botaoDeConfirmacao.exists()).toBeFalsy();
    done();
  });

  it('Renderiza um botao de cancelamento', async (done) => {
    opcoes.textoCancelar = 'CANCELAR';
    factory.dadoAProp(opcoes);
    await factory.montarComponente();

    const botaoDeConfirmacao = factory.componente.find('#cancelar');
    expect(botaoDeConfirmacao.exists()).toBeTruthy();
    expect(botaoDeConfirmacao.text()).toEqual('CANCELAR');
    done();
  });

  it('Não renderiza um botao de cancelamento', async (done) => {
    factory.dadoAProp(opcoes);
    await factory.montarComponente();

    const botaoDeConfirmacao = factory.componente.find('#cancelar');
    expect(botaoDeConfirmacao.exists()).toBeFalsy();
    done();
  });

  it('Snapshot', async (done) => {
    opcoes.titulo = 'titulo';
    opcoes.textoConfirmar = 'confirmar';
    opcoes.textoCancelar = 'cancelar';

    await factory.montarComponente();

    expect(factory.componente).toMatchSnapshot();
    done();
  });
});
