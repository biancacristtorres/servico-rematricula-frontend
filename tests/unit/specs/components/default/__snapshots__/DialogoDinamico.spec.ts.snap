// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente de Diálogo Dinâmico Snapshot 1`] = `
<vdialog-stub persistent="true" maxwidth="500" origin="center center" width="auto" scrollable="true" transition="dialog-transition">
  <vcard-stub tag="div" id="dialogo-dinamico">
    <vcardtitle-stub primarytitle="true" class="headline">

    </vcardtitle-stub>
    <vcardtext-stub>
      <div id="componente-dinamico-dummy"></div>
    </vcardtext-stub>
    <vcardactions-stub>
      <vspacer-stub></vspacer-stub>
      <!---->
      <!---->
    </vcardactions-stub>
  </vcard-stub>
</vdialog-stub>
`;
